apps/common/lib/utils.ex

```diff
@@ -0,0 +1,14 @@
+defmodule Common.Utils do
+  @moduledoc "Module for utilities used across belethor."
+
+  @doc """
+  macro to correctly log debug messages.
+
+  Calls `Logger.debug` in a way that makes dialyzer happy and does impact performance in production.
+  """
+  defmacro debug(msg) do
+    quote do
+      :ok = Logger.debug(fn -> unquote(msg) end)
```

Comment:
i _think_ this should be `_ =` not `:ok =` since it can technically return an error tuple

Reply:
depends should the code abort if the logging does not work?
with debug, I would say yes.

If you want to log an error and that logging fails, the code should abort / crash too.

Reply:
in that case, use a with statement, and raise when it fails to match :ok. something like
```elixir
with :ok = Logger.debug(fn -> unquote(msg) end) do
  :ok
else
  _ -> raise "Failed to log message!"
end
```